### ISSUE TITLE
Add Support for React-Native 74 and 75

### ISSUE THREAD

**Issue by nmongiya** [BabylonJS/BabylonReactNative#655]
**For questions, please use the [forum](https://forum.babylonjs.com)**.

**Is your feature request related to a problem? Please describe.**
React Native released version 74 in April and 75 on August 14, 2024. Can we please have support for that?

**Describe the solution you'd like**
I am not able to run the app with RN version 0.74.5

**Describe alternatives you've considered**
It compiles fine but crashes at runtime with the following error
`E  terminating due to uncaught exception of type std::logic_error: Not implemented`
 `A  Fatal signal 6 (SIGABRT), code -1 (SI_QUEUE) in tid 6035 (mqt_js), pid 5842 (xx.xx.xx)
`

**Additional context**

**Comment by CedricGuillemet** [comment]
I'm taking a look.

**Comment by CedricGuillemet** [comment]
0.74.2 is ok but 0.74.3 is not. Now searching for the breaking change.

**Comment by nmongiya** [comment]
You are right, I can confirm that I am able to run the app on 0.74.2

**Comment by nmongiya** [comment]
https://github.com/facebook/react-native/releases/tag/v0.74.3

**Comment by CedricGuillemet** [comment]
I could get a callstack:
```
/data/app/~~Unibl6ffkT11CnblQwHBjQ==/com.playground-RFNTdutP7QZS1IUYZa6Paw==/base.apk!libhermes.so (BuildId: 7f8ff6385b5fe455c03f7b48cdae12ba78b8c714)
2024-09-13 09:23:00.012 21147-21147 DEBUG                   pid-21147                            A        #01 pc 000000000034d87c  /data/app/~~Unibl6ffkT11CnblQwHBjQ==/com.playground-RFNTdutP7QZS1IUYZa6Paw==/base.apk!libBabylonNative.so (BabylonNative::Initialize(facebook::jsi::Runtime&, std::__ndk1::function<void (std::__ndk1::function<void ()>)>)+104) (BuildId: b431f252b69276ed8904a3dd383684bd86d07dd1)
2024-09-13 09:23:00.012 21147-21147 DEBUG                   pid-21147                            A        #02 pc 000000000034c5f8  /data/app/~~Unibl6ffkT11CnblQwHBjQ==/com.playground-RFNTdutP7QZS1IUYZa6Paw==/base.apk!libBabylonNative.so (Java_com_babylonreactnative_BabylonNativeInterop_00024BabylonNative_initialize+164) (BuildId: b431f252b69276ed8904a3dd383684bd86d07dd1)
2024-09-13 09:23:00.012 21147-21147 DEBUG                   pid-21147                            A        #03 pc 0000000000351230  /
```
tldr: it looks like an initialization issue with Hermes.

**Comment by CedricGuillemet** [comment]
@nmongiya I'm testing on Android for practical reason. Do you repro this crash on iOS as well?

**Comment by CedricGuillemet** [comment]
I can see a different version of hermes-android with the same version as RN but I can't find a package/build/changelog for that.
Also, I unzipped the .apk. `libhermes.so` file is different between both version but nothing striking when looking at symbols with objdump.

**Comment by CedricGuillemet** [comment]
The vertex buffer error is on the JS side and is not related to the RN version. It should happen with the web version as well.

**Comment by dmitryshostak** [comment]
@CedricGuillemet Hello, mate! Any updates? Is anyone working on adding the support? 👀

**Comment by okwasniewski** [comment]
I've also been getting crashes on 0.73 when working on this PR: #654 I thought its something related to my PR thankfully its not, 0.75 works as expected.

I've been getting the issue immediately after calling `BabylonNative::Initialize(*GetJSIRuntime(bridge), std::move(jsDispatcher));` which looks like an issue with accessing jsi runtime? 

There have been some changes around this here: https://github.com/reactwg/react-native-new-architecture/discussions/160 maybe some fixes didn't get backported to 0.73?

**Comment by CedricGuillemet** [comment]
@okwasniewski yes, there is an issue open for it : https://github.com/BabylonJS/BabylonReactNative/issues/655#issuecomment-2348225773
a new virtual method has been added between 0.74.2 and 0.74.3 so ABI is broken.
I didn't check 0.73. I was expecting this major version to work as 0.74.0 to 0.74.2 work

**Comment by antonsthb** [comment]
We experience the same issue using react-native 0.75.4 on both android and ios. Did you manage to find a workaround for this?

**Comment by mikevocalz** [comment]
any update?

**Comment by CedricGuillemet** [comment]
> any update?

Hi @mikevocalz , this is still on our roadmap but not much progress so far.

**Comment by johncobain** [comment]
any updates? i tried using in react-native 0.76.9 and i got `could not initialize babylon native React` on my app screen

**Comment by CedricGuillemet** [comment]
Fixed by https://github.com/BabylonJS/BabylonReactNative/pull/665
